### ISSUE TITLE
How to calculate the stellar mass for satellites?

### ISSUE THREAD

**Issue by WeakDog** [astropy/halotools#949]
I'm now using the Leauthaud et al. (2011) model, and know how to calculate the mean stellar mass of central galaxies. But I still have no idea how to calculate the stellar mass for satellites. Is there also a function for this?

**Comment by aphearin** [comment]
The Leauthaud et al. (2011) model is an HOD model, and by definition HOD satellites do not have a stellar mass, all they have is a "yes" or "no" specifying whether the satellite exceeds some luminosity threshold. If you need all galaxies to have a stellar mass, you will either need a CLF/CSMF model, or you can alternatively use models based on subhalos and use a stellar-to-halo-mass relation to map M* onto every (sub)halo.

**Comment by WeakDog** [comment]
Yes, I think so. We need the sub-halo information for the satellite mass. Thanks for your answering.

**Comment by aphearin** [comment]
If you do not have subhalo information available in your catalog, then you can use what is known as a Conditional Stellar Mass Function model (CSMF). What this specifies is the stellar mass function conditioned on host halo mass. In such a model, satellites _do_ have stellar mass. 

The `cacciato09` model in halotools can be used for this purpose. Strictly speaking, this model is actually a Conditional _Luminosity_ Function (CLF), but so far as the code is concerned, luminosity is just a label, and you can use this model equally well to model stellar mass instead. For example, @johannesulf used this model in this way in [his recent paper](https://arxiv.org/abs/1906.08680) on BOSS galaxies.

**Comment by WeakDog** [comment]
But in Leauthaud et al. 2011, there is already a model to calculate the stellar mass of satellites, like below,

<img width="317" alt="image" src="https://user-images.githubusercontent.com/30678982/66551149-d813de00-eb81-11e9-877d-f751a08c89ea.png">

Do you think if it's working?

**Comment by aphearin** [comment]
@WeakDog - did you get to the bottom of your question about satellite stellar masses? The equation you pointed to is related to internal parameters of the Leauthaud HOD model, nothing to do with assigning stellar masses to satellites.
